Add RLE transit TIF and fix `simplify = TRUE` logic to account for transit TIFs

erhla opened the pull request:
RE #56. I'd propose squashing this in if accepted since it took me a minute to figure out a few issues. 

This:

- Adds in the RLE tif into the transit tif calculations. (It was previously missing)
- Spins off the transit tif calculations into a nested function in tax bill. (not strictly necessary but it is more clear)
- Proposes adding a new version of `simplify=TRUE` for the `tax_bill` function called `simplify_transit`. I created this as an additional parameter but I would recommend that `simplify=TRUE` be replaced with `simplify_transit` since `simplify` is designed to mirror the agency line items on the tax bill but does not show agency line items on the tax bills in transit tifs as they actually are. AFAIK there shouldn't be any issues with replacing simplify with simplify_transit but I wanted to defer to you instead of proposing that directly. Let me know and I can make that change.
- Rename all the `RPM_` specific transit TIF variables to be named `transit_` since there are now two transit TIFs.
- update `final_tax_to_tif` to include `final_tax_to_dist`. The Clerk TIF revenue by tax code report includes the TIF revenue before either the CPS or non-CPS distributions are made from Transit TIFs. This means that if you sum the `final_tax_to_tif` it will align with the Clerk report.

This does not:

- Fix the issue identified related to how the actual tax bills calculate the agency line items which appear to be incorrect. The PTAXSIM calculations are technically correct but it appears that the Clerk/Treasurer are doing something a little different.

erhla committed:
hotfix for missing RLE tif

RLE tif is not calculated using the transit tif calculations. This is a hotfix which combines both transit tifs together and only leads to a small amount of error ~2%

erhla committed:
splitting the transit tif aspect into another function and renaming all rpm spec variables to be generic

erhla committed:
add new simplify parameter to show tax bills for transit tifs identical to tax bills

erhla committed:
only include cps tif row if exists

erhla committed:
lint/update docs

erhla committed:
cleanup readme and tests

erhla committed:
Update test-tax_bill.R

jeancochrane commented:
Thanks for this @erhla! I'm going to take a look early next week. These days @kyrasturgill gets final say over this repo, and she's out on vacation currently, so we won't be able to merge till the week of 8/11 at the earliest.

erhla commented:
I talked with the Treasurer today. They confirmed that the line items on the transit tax bills do not reflect the actual distributions to the agencies and that we should not try to replicate those calculations. They will send me the actual distribution percentages by tax code so I will verify that our calculations are correct compared to that.

jeancochrane reviewed (changes requested):
This looks awesome, thanks @erhla! Two chores that @kyrasturgill and I think you should do before we merge:

1. Consolidate `simplify` and `simplify_transit`
    * If we're understanding your suggestion correctly, then we agree with you that we should simply replace the `simplify` logic with `simplify_transit`, but continue using the parameter name `simplify`; if that's a misunderstanding of your suggestion, let us know and we can talk it through
3. Tweak unit tests to make sure they pass
    * Neither of us have inspected the unit tests in great detail yet, but it may make sense to add a test specific to the transit TIF simplification logic, since it's slightly different from the other simplification logic; we're open to that not being necessary if you think the existing `simplify` tests cover the new logic well enough, though

Once those changes are in, we'll do some more rigorous QC and then approve once all looks good.

erhla committed:
combine simplify / simplify_transit

erhla committed:
make unit tests pass

long story short the way the transit tif was shown on the tax bill has not been consistent over time so I've removed them from many of the unit tests

erhla committed:
oops

erhla committed:
change definition of final_tax_to_tif to include transit_tif_to_dist

erhla commented:
Now updated `final_tax_to_tif` to include all three components: example for 2023

<img width="1440" height="134" alt="image" src="https://github.com/user-attachments/assets/18fc09da-2da4-496e-9722-2834b44f3ecf" />

erhla committed:
cleanup and bump version number

jeancochrane commented:
@erhla I fixed a bug in #60 that I believe has been causing all of your CI checks to fail. Once you're ready to work on getting CI checks to pass, try merging in (or rebasing off of) the latest commit to the main branch of `ccao-data/ptaxsim` to fix that bug on your branch.

erhla committed:
fix simplify = false calculations for final tax, clarify docs

erhla committed:
new test for simplify = false to align with total tax code taxes collected

erhla committed:
Merge remote-tracking branch 'upstream/master'

erhla committed:
fix tests

erhla committed:
fix tests

erhla commented:
Alright I think all looks good here. I added a few tests to ensure that the transit tif / simplify changes calculate things correctly. I was thinking about adding a vignette but I feel like transit tifs are not worth their own vignette and also adding them into the tinkering with tifs vignette feels kind of random.

jeancochrane commented on R/tax_bill.R:
```diff
@@ -58,10 +58,16 @@
-#'   appear on a real tax bill. Additionally, collapses the TIF output column
-#'   \code{final_tax_to_tif} to a line-item, similar to the format on a real
-#'   tax bill.
+#' @param simplify Default \code{TRUE}. Boolean to keep only the columns
+#'   that appear on a real tax bill
+#'   which also includes the "Board of Education - from Transit TIF" line.
+#'   Additionally, collapses the TIF output column \code{final_tax_to_tif}
+#'   to a line-item, similar to the format on a real tax bill. Note: In order to
```
**[Nitpick, optional]** It took me a while to parse what these docs were trying to say, so here's an attempt at an improvement that I think will be clearer to someone who is using this package for the first time:

```suggestion
#' @param simplify Default \code{TRUE}. Boolean to keep only the columns
#'   that appear on a real tax bill.
#'   If the parcel is in a TIF, simplified output will collapse the TIF output
#'   column \code{final_tax_to_tif} to a line item, similar to the format on a
#'   real tax bill; further, if the parcel is in a transit TIF, simplified
#'   output will also collapse the column \code{transit_tif_to_cps} to a line
#'   item called "Board of Education - from Transit TIF". Note: In order to
#'   more completely show information for transit TIF districts both
#'   \code{final_tax_to_tif} and \code{final_tax_to_dist} include
#'   \code{transit_tif_to_dist}. Therefore, to calculate final total taxes
#'   when \code{FALSE}, use
#'   \code{final_tax_to_tif + final_tax_to_dist - transit_tif_to_dist}.
```

jeancochrane commented on tests/testthat/test-tax_bill.R:
```diff
@@ -145,16 +147,28 @@ test_that("returned amount/output correct for all sample bills", {
+    select(year, pin, agency_num, final_tax)
+  transit_bills <- base_bills %>%
+    count(pin, agency_num) %>%
+    filter(n > 1)
```
**[Nitpick, optional]** It took me a minute to figure out why this filter would return only transit bills, so a comment for posterity:

```suggestion
    # PINs in transit TIFs will have two line items with the CPS agency number,
	# one being the TIF distribution
    filter(n > 1)
```

jeancochrane commented on tests/testthat/test-tax_bill.R:
```diff
@@ -145,16 +147,28 @@ test_that("returned amount/output correct for all sample bills", {
     tax_bill(sum_dt$year, sum_dt$pin, simplify = TRUE) %>%
       select(year, pin, agency_num, final_tax) %>%
+      filter(!pin %in% transit_bills$pin) %>%
+      group_by(year, pin, agency_num) %>%
+      summarize(final_tax = sum(final_tax)) %>%
+      # combine cps transit tif rows w/ tif row
```
**[Nitpick, optional]** I think moving this comment to the top of the grouping block will make it clearer:

```suggestion
      # combine cps transit tif rows w/ tif row
      group_by(year, pin, agency_num) %>%
      summarize(final_tax = sum(final_tax)) %>%
```

But also, do we even need to do this, given that we're filtering for `!pin %in% transit_bills$pin` above?

jeancochrane reviewed (approved):
Really nice work here! No serious concerns from my end, just a bunch of clarifying questions and small nitpicks that I think will make the code easier to read for other rookies like me. We're at the nitpick stage, so I'm happy to do the legwork of resolving all of our remaining nits and getting the PR merged if you'd prefer that.

erhla committed:
Update R/tax_bill.R

remove unnecessary comment

Co-authored-by: Jean Cochrane <jeancochrane@users.noreply.github.com>

erhla committed:
Update tests/testthat/test-accuracy.R

Co-authored-by: Jean Cochrane <jeancochrane@users.noreply.github.com>

erhla committed:
Update tests/testthat/test-tax_bill.R

Co-authored-by: Jean Cochrane <jeancochrane@users.noreply.github.com>

erhla committed:
Update R/tax_bill.R

Co-authored-by: Jean Cochrane <jeancochrane@users.noreply.github.com>

erhla committed:
final review comments

erhla commented:
Thanks! I think all the comments both of you made are clear to me.

I've enjoyed this. Normally I'm just a team of one so this was really fun for me, but please squash all my commits.

Now we just have to monitor how the Treasurer is going to update this for the 2024 bills.

jeancochrane commented:
@erhla It seems like you resolved a few of my questions without adding answers, I wonder if you added comments but then didn't submit the review?

> I've enjoyed this. Normally I'm just a team of one so this was really fun for me, but please squash all my commits.

Agreed, it's also a pleasure for us to work with someone externally who has the technical chops to make a contribution like this!

erhla commented:
>  It seems like you resolved a few of my questions without adding answers, I wonder if you added comments but then didn't submit the review?

Ah for those I accepted your suggestion as is and then resolved your question. I haven't used github like this before so let me know if there's something different I should have done.

jeancochrane commented on R/tax_bill.R:
```diff
@@ -58,10 +58,23 @@
+#'   \code{final_tax_to_tif} and \code{final_tax_to_dist} include
+#'   \code{transit_tif_to_dist}. Therefore, to calculate final total taxes
+#'   when \code{FALSE}, use
+#'   \code{final_tax_to_tif + final_tax_to_dist - transit_tif_to_dist}.
+#'   more completely show information for transit TIF districts both
+#'   \code{final_tax_to_tif} and \code{final_tax_to_dist} include
+#'   \code{transit_tif_to_dist}. Therefore, to calculate final total taxes
+#'   when \code{FALSE}, use
```
**[Nitpick, required]** Looks like we accidentally duplicated some lines of this docstring:

```suggestion
```

We'll also need to regenerate the autogenerated `.Rd` docs after we make this change.

erhla committed:
Update R/tax_bill.R

Co-authored-by: Jean Cochrane <jeancochrane@users.noreply.github.com>

erhla committed:
update docs

erhla committed:
Update test-tax_bill.R

erhla reviewed (commented):
Turns out I didn't realize you have to commit the comments! So here's my answers (belatedly).

erhla committed:
Merge branch 'master' into master

erhla commented:
I'm looking at the new tax bills for tax year 2024 and the treasurer has changed so many things tragically. I'm hoping they might clean it up more but initially it looks like it is much less clear. I think we should ideally merge this in for ty23 and it probably makes sense to show the information in the most clear way we believe for ty24 and not how the treasurer is doing it. I want to see some of the tif reports from the clerk first to see if the calculations are even being doing correctly.

erhla commented:
<img width="1777" height="1123" alt="image" src="https://github.com/user-attachments/assets/56b498d9-a315-4f09-a355-e672241277da" />

Here is an example. Note that the Board of Education TIF and regular levy is now combined into one item confusingly labeled board of education TIF RPM1

jeancochrane commented on R/tax_bill.R:
```diff
@@ -299,16 +351,19 @@ tax_bill <- function(year_vec,
-  } else {
+  }
+
+  if (!simplify) {
```
**[Nitpick, required]** For the sake of cleanliness, let's revert to `else` here, since it's a more standard way to mark the false branch of a conditional expression:


```suggestion
  } else {
```

jeancochrane reviewed (approved):
Thanks for your patience @erhla! Pending your resolution of one final nitpick from my end, this is good to merge. I'll look out for a commit from you on that front and merge once it lands.

Heads up that I tweaked the title of the PR to be more descriptive -- feel free to change it if you think I've misrepresented the PR at all.

> I'm looking at the new tax bills for tax year 2024 and the treasurer has changed so many things tragically. I'm hoping they might clean it up more but initially it looks like it is much less clear. I think we should ideally merge this in for ty23 and it probably makes sense to show the information in the most clear way we believe for ty24 and not how the treasurer is doing it. I want to see some of the tif reports from the clerk first to see if the calculations are even being doing correctly.

Agreed. I added a quick note about this to our [2024 refresh omnibus issue](https://github.com/ccao-data/ptaxsim/issues/59) to make sure we don't lose track of it. In the meantime, I think the diff in this PR is a big improvement.

erhla committed:
Update R/tax_bill.R

simplify is true/false boolean again

Co-authored-by: Jean Cochrane <jeancochrane@users.noreply.github.com>

erhla commented:
Amazing, thank you! re #59, based on what the Treasurer said it looks like TIF calculations are done by pin (and the pin's frozen eav at the time the TIF was created) now instead of tax code which is going to be interesting to say the least. So I think we will permanently need to calculate TY23 and earlier tax bills and TY24 and later bills with different code.

jeancochrane commented:
> based on what the Treasurer said it looks like TIF calculations are done by pin (and the pin's frozen eav at the time the TIF was created) now instead of tax code which is going to be interesting to say the least. So I think we will permanently need to calculate TY23 and earlier tax bills and TY24 and later bills with different code.

Agreed, I just updated the issue text to reflect this.

Thanks again for your hard work getting this landed!
